### PR TITLE
Sample on root span level

### PR DESCRIPTION
**Pull request by sentrivana**
- for spans that have a local parent (i.e., non-root spans/transactions), just inherit the sampling decision from the parent
- update some tests

Closes https://github.com/getsentry/sentry-python/issues/3766

### CODE REVIEW

File: sentry_sdk/tracing.py [hunk 1]
```diff
@@ -1212,6 +1212,7 @@ def __init__(
         attributes=None,  # type: OTelSpanAttributes
         only_if_parent=False,  # type: bool
         otel_span=None,  # type: Optional[OtelSpan]
+        parent_sampled=None,  # type: Optional[bool]
```
Comment (sentrivana) [inline sentry_sdk/tracing.py:1215]: Not sure if we want to continue supporting this arg, but added it for now

File: sentry_sdk/integrations/opentelemetry/sampler.py [hunk 2]
```diff
@@ -120,33 +120,44 @@ def should_sample(
...
+                    f"[Tracing] Ignoring sampled param for non-root span {name}"
+                )
 
         sample_rate = None
 
         # Check if there is a traces_sampler
         # Traces_sampler is responsible to check parent sampled to have full transactions.
         has_traces_sampler = callable(client.options.get("traces_sampler"))
-        if has_traces_sampler:
+        parent_sampled = attributes.get(SentrySpanAttribute.CUSTOM_PARENT_SAMPLED)
```
[the remark this answers is not in the document]
Comment (sentrivana) [reply sentry_sdk/integrations/opentelemetry/sampler.py]: removed altogether

**Comment by sentrivana** [comment]
@sl0thentr0py Removed `parent_sampled` and updated the tests a bit more, please recheck

File: tests/tracing/test_sampling.py [hunk 3]
```diff
@@ -177,38 +183,32 @@ def test_inherits_parent_sampling_decision_when_traces_sampler_undefined(
     mock_random_value = 0.25 if parent_sampling_decision is False else 0.75
 
     with mock.patch.object(random, "random", return_value=mock_random_value):
-        transaction = start_transaction(
-            name="dogpark", parent_sampled=parent_sampling_decision
-        )
-        assert transaction.sampled is parent_sampling_decision
+        with start_span(name="catpark", sampled=parent_sampling_decision):
```
Comment (sl0thentr0py) [inline tests/tracing/test_sampling.py:186]: this test should mirror the one above  where you added the continue_trace

**Review by sl0thentr0py — CHANGES_REQUESTED** [review]

**Review by sl0thentr0py — APPROVED** [review]

**Pull request merged**